### PR TITLE
Patched Music commands

### PR DESCRIPTION
**Pull request by NapkinsXYZ**
Updated Shuffle, Pause, Skip, Queue, and Resume
Added more aliases for the commands (Queue, Shuffle, etc.)

### CODE REVIEW

File: commands/music.py [hunk 1]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
         await ctx.voice_state.stop()
         del self.voice_states[ctx.guild.id]
 
-    @commands.command(name='volume')
-    async def _volume(self, ctx: commands.Context, *, volume: int):
-        """Sets the volume of the player."""
+    @commands.command(name='volume', aliases=['vol'])
+    async def change_volume(self, ctx, *, vol: float):
```
Comment (wakfi) [inline commands/music.py:329]: I thought this didn't have any affect since users can set volume locally, and was being removed. Has this changed, or does it work now?

File: commands/music.py [hunk 2]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
-    async def _now(self, ctx: commands.Context):
-        """Displays the currently playing song."""
+        player = self.get_player(ctx)
 
-        await ctx.send(embed=ctx.voice_state.current.create_embed())
+        if vc.source:
+            vc.source.volume = vol / 100
+
+        player.volume = vol / 100
+        await ctx.send(f'**`{ctx.author}`**: Set the volume to **{vol}%**')
```
Comment (wakfi) [inline commands/music.py:350]: To me this feels like it will look weird with "Set" having an uppercase 'S' as it isn't the first word if the sentence. Depends on if this is supposed to operate as an announcement that the user took the action, a statement on behalf of the user, or a reply to the user

File: commands/music.py [hunk 3]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
+        if not vc or not vc.is_playing():
+            return await ctx.send('I am not currently playing anything!', delete_after=20)
+        elif vc.is_paused():
+            return
 
-        if not ctx.voice_state.is_playing and ctx.voice_state.voice.is_playing():
-            ctx.voice_state.voice.pause()
-            await ctx.message.add_reaction('⏯')
+        vc.pause()
+        await ctx.send(f'**`{ctx.author}:`** Paused the song!')
```
Comment (wakfi) [inline commands/music.py:363]: Same note as with "Set"

File: commands/music.py [hunk 4]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
+        if not vc or not vc.is_connected():
+            return await ctx.send('I am not currently playing anything!', delete_after=20)
+        elif not vc.is_paused():
+            return
 
-        if not ctx.voice_state.is_playing and ctx.voice_state.voice.is_paused():
-            ctx.voice_state.voice.resume()
-            await ctx.message.add_reaction('⏯')
+        vc.resume()
+        await ctx.send(f'**`{ctx.author}:`** Resumed the song!')
```
Comment (wakfi) [inline commands/music.py:376]: Same note as with "Set"

File: commands/music.py [hunk 5]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
-    @commands.has_permissions(manage_guild=True)
-    async def _pause(self, ctx: commands.Context):
-        """Pauses the currently playing song."""
+    async def pause_(self, ctx):
+        """Pause the currently playing song."""
+        vc = ctx.voice_client
+
+        if not vc or not vc.is_playing():
+            return await ctx.send('I am not currently playing anything!', delete_after=20)
+        elif vc.is_paused():
```
Comment (wakfi) [inline commands/music.py:359]: This can just be an `if`

File: commands/music.py [hunk 6]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
-    @commands.has_permissions(manage_guild=True)
-    async def _resume(self, ctx: commands.Context):
-        """Resumes a currently paused song."""
+    async def resume_(self, ctx):
+        """Resume the currently paused song."""
+        vc = ctx.voice_client
+
+        if not vc or not vc.is_connected():
+            return await ctx.send('I am not currently playing anything!', delete_after=20)
+        elif not vc.is_paused():
```
Comment (wakfi) [inline commands/music.py:372]: This can just be an `if`

File: commands/music.py [hunk 7]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
+            return await ctx.send('Please enter a value between 1 and 100.')
 
-    @commands.command(name='now', aliases=['current', 'playing'])
-    async def _now(self, ctx: commands.Context):
-        """Displays the currently playing song."""
+        player = self.get_player(ctx)
 
-        await ctx.send(embed=ctx.voice_state.current.create_embed())
+        if vc.source:
+            vc.source.volume = vol / 100
```
Comment (wakfi) [inline commands/music.py:347]: (assuming volume changes have an impact in the client) Does this have a stacking effect with the volume change below it? Does that cause it to change too much?

File: commands/music.py (hunk 1 above)
Comment (NapkinsXYZ) [reply commands/music.py:329]: It didn't seem to work even after changing it. I decided to leave it there to see if we would try to get it working or just abandon it. I personally think it's unnecessary to have and am with the idea of removing it.

Comment (wakfi) [reply commands/music.py:329]: You're running this feature implementation. If it doesn't do anything and you're in favor of taking it out, I don't see any reason to keep it. Particularly since it isn't needed anyways.

File: commands/music.py (hunk 5 above)
Comment (wakfi) [reply commands/music.py:359]: This is referencing the `elif`, it was only supposed to comment the one line. Same with the matching comments below.

**Comment by wakfi** [comment]
Piggybacking a deploy change onto this PR

File: commands/music.py (hunk 1 above)
Comment (wakfi) [inline commands/music.py:329]:
```suggestion
```

File: commands/music.py [hunk 8]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
         await ctx.voice_state.stop()
         del self.voice_states[ctx.guild.id]
 
-    @commands.command(name='volume')
-    async def _volume(self, ctx: commands.Context, *, volume: int):
-        """Sets the volume of the player."""
+    @commands.command(name='volume', aliases=['vol'])
+    async def change_volume(self, ctx, *, vol: float):
+        """Change the player volume.
+        Parameters
```
Comment (wakfi) [inline commands/music.py:332]:
```suggestion
```

File: commands/music.py [hunk 9]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
 
-        if not ctx.voice_state.is_playing:
-            return await ctx.send('Nothing being played at the moment.')
+        volume: float or int [Required]
+            The volume to set the player to in percentage. This must be between 1 and 100.
+        """
+        vc = ctx.voice_client
```
Comment (wakfi) [inline commands/music.py:337]:
```suggestion
```

File: commands/music.py [hunk 10]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
 
-        if 0 > volume > 100:
-            return await ctx.send('Volume must be between 0 and 100')
+        if not vc or not vc.is_connected():
+            return await ctx.send('I am not currently connected to voice!', delete_after=20)
```
Comment (wakfi) [inline commands/music.py:340]:
```suggestion
```

File: commands/music.py [hunk 11]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
 
-        ctx.voice_state.volume = volume / 100
-        await ctx.send('Volume of the player set to {}%'.format(volume))
+        if not 0 < vol < 101:
+            return await ctx.send('Please enter a value between 1 and 100.')
```
Comment (wakfi) [inline commands/music.py:343]:
```suggestion
```

File: commands/music.py [hunk 12]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
 
-    @commands.command(name='now', aliases=['current', 'playing'])
-    async def _now(self, ctx: commands.Context):
-        """Displays the currently playing song."""
+        player = self.get_player(ctx)
```
Comment (wakfi) [inline commands/music.py:345]:
```suggestion
```

File: commands/music.py (hunk 2 above)
Comment (wakfi) [inline commands/music.py:350]:
```suggestion
```

File: commands/music.py (hunk 5 above)
Comment (wakfi) [inline commands/music.py:359]:
```suggestion
        if vc.is_paused():
```

File: commands/music.py (hunk 7 above)
Comment (wakfi) [reply commands/music.py:347]: This is resolved if volume is removed

File: commands/music.py (hunk 6 above)
Comment (wakfi) [inline commands/music.py:372]:
```suggestion
        if not vc.is_paused():
```

File: commands/music.py (hunk 4 above)
Comment (wakfi) [inline commands/music.py:376]:
```suggestion
        await ctx.send(f'**`{ctx.author}:`** resumed the song!')
```

File: commands/music.py (hunk 3 above)
Comment (wakfi) [inline commands/music.py:363]:
```suggestion
        await ctx.send(f'**`{ctx.author}:`** paused the song!')
```

File: commands/music.py (hunk 2 above)
Comment (wakfi) [reply commands/music.py:350]: This is resolved if volume is removed

File: commands/music.py [hunk 13]
```diff
@@ -308,7 +314,7 @@ async def _summon(self, ctx: commands.Context, *, channel: discord.VoiceChannel
 
         ctx.voice_state.voice = await destination.connect()
 
-    @commands.command(name='leave', aliases=['disconnect'])
+    @commands.command(name='leave', aliases=['disconnect', 'quit'])
     @commands.has_permissions(manage_guild=True)
```
Comment (wakfi) [inline commands/music.py:318]:
```suggestion
    @commands.has_permissions(move_members=True)
```
Manage guild seems a bit excessive to me and probably won't line up much with the final permission structure. Move members gives the disconnect ability IIRC, so I feel like this would make sense for now

**Review by wakfi — COMMENTED** [review]
Updated suggestions to be actual suggestions, you should now be able to opt to include or not include them directly

**Comment by wakfi** [comment]
@NapkinsXYZ what is the status on this? Have the suggestions been addressed in some way? You can just pick to accept or decline each of them I think.

File: commands/music.py [hunk 14]
```diff
@@ -319,53 +325,67 @@ async def _leave(self, ctx: commands.Context):
...
-            return await ctx.send('Volume must be between 0 and 100')
+        if not 0 < vol < 101:
+            return await ctx.send('Please enter a value between 1 and 100.')
 
-        ctx.voice_state.volume = volume / 100
-        await ctx.send('Volume of the player set to {}%'.format(volume))
+        player = self.get_player(ctx)
 
-    @commands.command(name='now', aliases=['current', 'playing'])
-    async def _now(self, ctx: commands.Context):
```
Comment (wakfi) [inline commands/music.py:336]: Why was this removed?

**Review by DaRealCodeWritten — APPROVED** [review]
Looks good to me, lets pray for no odd issues when this goes into develop
